src/fides/api/task/graph_task.py

```diff
@@ -603,12 +603,19 @@ def erasure_request(
         *erasure_prereqs: int,  # TODO Remove when we stop support for DSR 2.0. DSR 3.0 enforces with downstream_tasks.
     ) -> int:
         """Run erasure request"""
+
         # if there is no primary key specified in the graph node configuration
         # note this in the execution log and perform no erasures on this node
-        if not self.execution_node.collection.contains_field(lambda f: f.primary_key):
+        if (
+            self.connector.requires_primary_keys
+            and not self.execution_node.collection.contains_field(
+                lambda f: f.primary_key
+            )
+        ):
```

Comment:
Key change - This is where I added the additional check for `requires_primary_keys`